std/algorithm/comparison.d

```diff
@@ -724,6 +776,60 @@ if (isInputRange!R1 && isInputRange!R2)
     assert(result > 0);
 }
 
+///
+@safe unittest
+{
+    int result;
+
+    result = cmp!"a > b"("abc", "abc");
+    assert(result == 0);
+    result = cmp!"a > b"("", "");
+    assert(result == 0);
+    result = cmp!"a > b"("abc", "abcd");
+    assert(result > 0);
```

Comment:
This behavior is a bug. r1 is a prefix of r2, so the result should be negative even if we are comparing in reverse alphabetical order, but `cmp` is using `pred` to compared the lengths! I have no sense of whether anyone might have been relying on this for reverse order comparison. It is a violation of the documentation and it only works for strings. Should it be fixed?

Reply:
That's terrible. I'd say open a bug with this repro then mark it as fixed by this PR. That way we'll have it in the changelog.